test/client-unit/components/bhFindInvoice.spec.js

```diff
@@ -0,0 +1,217 @@
+/* eslint no-unused-expressions:off */
+/* global inject, expect, chai */
+
+describe('bhFindInvoice', bhFindInvoiceTests);
+
+function bhFindInvoiceTests() {
+  const patient = {
+    uuid              : '234c51ae-efcc-4238-98c6-f402bfb39866',
+    reference         : 'TPA17',
+    dob               : '1993-04-25T23:00:00.000Z',
+    first_name        : 'Brandon',
+    middle_name       : 'Hope',
+    last_name         : 'Spike',
+    registration_date : '2018-05-10T23:00:000Z',
+    sex               : 'F',
+    invoice_reference : 'IV.TPA.001',
+  };
+
+  const invoice = {
+    uuid : 'abcc51ae-efcc-4238-98c6-f402bfb39866',
+    reference : 'IV.TPA.001',
+    patient_uuid : '234c51ae-efcc-4238-98c6-f402bfb39866',
+  };
+
+  const template = `
+    <bh-find-invoice
+      patient-uuid="patientUuid"
+      on-search-complete="callback(record)">
+    </bh-find-invoice>
+  `;
+
+  // make sure the modules are correctly loaded.
+  beforeEach(module(
+    'pascalprecht.translate',
+    'ngStorage',
+    'ui.bootstrap',
+    'angularMoment',
+    'tmh.dynamicLocale',
+    'bhima.services',
+    'bhima.components',
+    'bhima.constants',
+    'templates',
+  ));
+
+  let $scope;
+  let $compile;
+  let $controller;
+  let $httpBackend;
+  let element;
+  let Q;
+  let PatientInvoice;
+  let Notify;
+
+  // find element function without jquery dependencies
+  const find = (elm, selector) => elm[0].querySelector(selector);
+
+  // inject dependencies
+  beforeEach(inject((_$rootScope_, _$compile_, _$httpBackend_, _$q_, _PatientInvoiceService_, _$window_, _NotifyService_) => {
+    $compile = _$compile_;
+    $scope = _$rootScope_.$new();
+    $httpBackend = _$httpBackend_;
+    PatientInvoice = _PatientInvoiceService_;
+    Notify = _NotifyService_;
+    Q = _$q_;
+
+    element = angular.element(template);
+    element = $compile(element)($scope);
+
+    // overwrite the getElementById function
+    _$window_.document.getElementById = (id) => find(element, `#${id}`);
+
+    // overwrite the notification errorhandler and track it
+    Notify.handleError = () => {};
+    Notify.handleError = chai.spy();
+
+    $scope.patientUuid = '234c51ae-efcc-4238-98c6-f402bfb39866';
+    $scope.callback = chai.spy();
+    $scope.$digest();
+  }));
+
+  // make sure $httpBackend is clean after each test completes
+  afterEach(() => {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('#bhFindInvoice component exists and it is compiled', () => {
+    const input = find(element, '#search-input');
+    expect(angular.element(input).length).to.equal(1);
+
+    const button = find(element, '#search-button');
+    expect(angular.element(button).length).to.equal(1);
+  });
+
+  it('Runs the search method when the ENTER key is pressed', () => {
```

Comment:
These are really great tests, @mbayopanda .  The descriptions are clear, each test is small and self-contained.  Really great work!